types/encoding.go

```diff
@@ -136,8 +140,15 @@ func (d *Decoder) Read(p []byte) (int, error) {
 		if want > len(d.buf) {
 			want = len(d.buf)
 		}
-		var read int
-		read, d.err = io.ReadFull(&d.lr, d.buf[:want])
```

Comment:
Since the buffer wasn't cleared when a read error happened, future calls to `ReadPrefix` would LE decode whatever was in the buffer. Occasionally a massive value would be read in the wrong spot causing massive allocations and OOM.